lib/trino-hdfs/pom.xml

```diff
@@ -225,4 +225,42 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <profiles>
+        <profile>
+            <id>default</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <excludes>
+                                <exclude>**/TestFileSystemCache.java</exclude>
+                            </excludes>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>test-hdfs-cache</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <includes>
+                                <include>**/TestFileSystemCache.java</include>
```

Comment:
Why is separate profile needed? do you maybe understand why https://github.com/trinodb/trino/pull/17266 was not sufficient?

Reply:
One problem i see with #17266 is that the thread count is limited only for trino-hdfs module. There is a possibility that the tests in `trino-hdfs` and `trino-hive-hadoop2` can run parallelly in separate threads in the same JVM interacting with same global `TrinoFileSystemCache`. This can cause `testFileSystemCacheConcurrency()` to fail.

Reply:
> There is a possibility that the tests in `trino-hdfs` and `trino-hive-hadoop2` can run parallelly in separate threads in the same JVM interacting with same global `TrinoFileSystemCache`.

i'd hope different modules do not share same JVM for test execution.
I think maven surefire plugin starts another JVM for test run.

Reply:
I was able to spot an interaction from another thread that invoked `getFileSystem()` while the tests from `TestFileSystemCache` were running (From https://github.com/jitheshtr/trino/actions/runs/4964921803/jobs/8885392660?pr=6) -
```
2023-05-12T21:50:19.774-0600 SEVERE Could not process download requests
java.lang.RuntimeException: Saw user runner with null proxy user
	at io.trino.hdfs.TrinoFileSystemCache.createFileSystemKey(TrinoFileSystemCache.java:259)
	at io.trino.hdfs.TrinoFileSystemCache.getInternal(TrinoFileSystemCache.java:115)
	at io.trino.hdfs.TrinoFileSystemCache.get(TrinoFileSystemCache.java:94)
	at org.apache.hadoop.fs.ForwardingFileSystemCache.get(ForwardingFileSystemCache.java:38)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:479)
	at com.qubole.rubix.bookkeeper.FileDownloader.getFileDownloadRequestChains(FileDownloader.java:109)
	at com.qubole.rubix.bookkeeper.RemoteFetchProcessor.processRequest(RemoteFetchProcessor.java:155)
	at com.qubole.rubix.bookkeeper.RemoteFetchProcessor.runOneIteration(RemoteFetchProcessor.java:144)
	at com.google.shaded.shaded.common.util.concurrent.AbstractScheduledService$ServiceDelegate$Task.run(AbstractScheduledService.java:193)
	at com.google.shaded.shaded.common.util.concurrent.Callables$4.run(Callables.java:119)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

This exception was triggered by the code changes here - https://github.com/jitheshtr/trino/pull/6/commits/a1042a50953c512248e64696e00c11a266ca259a

Reply:
>i'd hope different modules do not share same JVM for test execution.
I think maven surefire plugin starts another JVM for test run.

You are right, from https://maven.apache.org/surefire/maven-surefire-plugin/examples/fork-options-and-parallel-execution.html it should invoke different modules in different JVMs -
> The default setting is forkCount=1/reuseForks=true, which means that maven-surefire-plugin creates one new JVM process to execute all tests in one Maven module.

Looking at above capture, simultaneous use of `TrinoFileSystemCache` from a different thread is happening from Rubix code running in the same JVM.

Reply:
let's capture the motivation (ie something mentioning "rubix") in the commit message AND a code comment in the profile definition inside the pom file